controller/appcontroller.go

```diff
@@ -487,16 +394,33 @@ func (ctrl *ApplicationController) finalizeApplicationDeletion(app *appv1.Applic
 		return err
 	}
 	config := clst.RESTConfig()
-	err = kube.DeleteResourcesWithLabel(config, app.Spec.Destination.Namespace, common.LabelApplicationName, app.Name)
+	objsMap, err := ctrl.stateCache.GetControlledLiveObjs(app, []*unstructured.Unstructured{})
 	if err != nil {
 		return err
 	}
-	objs, err := kube.GetResourcesWithLabel(config, app.Spec.Destination.Namespace, common.LabelApplicationName, app.Name)
+	objs := make([]*unstructured.Unstructured, 0)
+	for k := range objsMap {
+		objs = append(objs, objsMap[k])
+	}
+	err = util.RunAllAsync(len(objs), func(i int) error {
+		obj := objs[i]
+		return ctrl.kubectl.DeleteResource(config, obj.GroupVersionKind(), obj.GetName(), obj.GetNamespace())
+	})
 	if err != nil {
 		return err
 	}
-	if len(objs) > 0 {
-		logCtx.Info("%d objects remaining for deletion", len(objs))
+	for _, obj := range objs {
+		err := ctrl.kubectl.DeleteResource(config, obj.GroupVersionKind(), obj.GetName(), obj.GetNamespace())
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Why do you have a second iteration to delete the resources sequentially, when you just did it async?

Reply:
Forgot to delete the second iteration. Removed: https://github.com/argoproj/argo-cd/pull/840/commits/a842eb3bb31f378c37866ea5deba3efc150d4341